### PR TITLE
Built-in Sandbox: add more unit tests for dragonball

### PR DESCRIPTION
**Pull request by wllenyj**
Added more unit test for vmm actions.
Added dragonball components to Makefile.
Fixed clippy warning.

### CODE REVIEW

File: src/dragonball/src/api/v1/vmm_action.rs [hunk 1]
```diff
@@ -634,3 +634,626 @@ fn handle_cpu_topology(
 
     Ok(cpu_topology)
 }
+#[cfg(test)]
```
Comment (liubin) [inline src/dragonball/src/api/v1/vmm_action.rs:637]: Nit: Add a blank line between code and test mod.

File: src/dragonball/src/api/v1/vmm_action.rs [hunk 2]
```diff
@@ -634,3 +634,626 @@ fn handle_cpu_topology(
...
+        let mut event_mgr = EventManager::new(&vmm, epoll_mgr).unwrap();
+        let mut v = vmm.lock().unwrap();
+
+        let vm = v.get_vm_mut().unwrap();
+        vm.set_instance_state(vm_state);
+
+        to_vmm.send(Box::new(request)).unwrap();
+        assert!(vservice.run_vmm_action(&mut v, &mut event_mgr).is_ok());
+
+        let response = match from_vmm.try_recv() {
```
Comment (liubin) [inline src/dragonball/src/api/v1/vmm_action.rs:669]: How about:

```rust
let response = from_vmm.try_recv();
assert!(response.is_ok());
check_success(response.unwrap());
```

File: src/dragonball/src/api/v1/vmm_action.rs [hunk 3]
```diff
@@ -634,3 +634,626 @@ fn handle_cpu_topology(
...
+    }
+
+    #[test]
+    fn test_vmm_action_set_vm_configuration() {
+        skip_if_not_root!();
+
+        // invalid state
+        let req = VmmAction::SetVmConfiguration(VmConfigInfo::default());
+        check_request(req, InstanceState::Running, |result| {
+            if let Err(VmmActionError::MachineConfig(VmConfigError::UpdateNotAllowedPostBoot)) =
```
Comment (liubin) [inline src/dragonball/src/api/v1/vmm_action.rs:767]: Is the `if let Err(xxx) = result` needed? can this be simplified by:

```diff
-             if let Err(VmmActionError::MachineConfig(VmConfigError::UpdateNotAllowedPostBoot)) =
-                 result
-             {
-                 let err_string = format!("{}", result.unwrap_err());
-                 let expected_err = String::from(
-                     "failed to set configuration for the VM: \
-                     update operation is not allowed after boot",
-                 );
-                 assert_eq!(err_string, expected_err);
-             } else {
-                 panic!("unexpect");
-             }
---
+             assert!(result.is_err());
+             let err_string = format!("{}", result.unwrap_err());
+             let expected_err = String::from(
+                 "failed to set configuration for the VM: \
+                 update operation is not allowed after boot",
+             );
+             assert_eq!(err_string, expected_err);

```

File: src/dragonball/src/api/v1/vmm_action.rs [hunk 4]
```diff
@@ -634,3 +634,626 @@ fn handle_cpu_topology(
...
+                );
+                assert_eq!(err_string, expected_err);
+            } else {
+                panic!();
+            }
+        });
+
+        // invalid guest_cid
+        let req = VmmAction::InsertVsockDevice(VsockDeviceConfigInfo::default());
+        check_request(req, InstanceState::Uninitialized, |result| {
```
Comment (liubin) [inline src/dragonball/src/api/v1/vmm_action.rs:1236]: All calling of `check_request` have the same pattern, so it is suitable for [table driven test](https://github.com/kata-containers/kata-containers/blob/f30fe86dc1f53b5e964e6f54de28a0ce0a818786/docs/Unit-Test-Advice.md#rust-table-driven-tests), suggestion fix this in this PR or another separated PR.

**Review by liubin — APPROVED** [review]

**Comment by liubin** [comment]
@wllenyj Do you have time to fix the static check error?

File: src/dragonball/src/api/v1/vmm_action.rs (hunk 1 above)
Comment (wllenyj) [reply src/dragonball/src/api/v1/vmm_action.rs:637]: fixed

File: src/dragonball/src/api/v1/vmm_action.rs (hunk 2 above)
Comment (wllenyj) [reply src/dragonball/src/api/v1/vmm_action.rs:669]: fixed

File: src/dragonball/src/api/v1/vmm_action.rs (hunk 3 above)
Comment (wllenyj) [reply src/dragonball/src/api/v1/vmm_action.rs:767]: fixed.

File: Makefile [hunk 5]
```diff
@@ -10,6 +10,7 @@ COMPONENTS += libs
 COMPONENTS += agent
 COMPONENTS += runtime
 COMPONENTS += runtime-rs
+COMPONENTS += dragonball
```
[the remark this answers is not in the document]
Comment (wllenyj) [reply Makefile:13]: fixed

**Comment by wllenyj** [comment]
> @wllenyj Do you have time to fix the static check error?

Attempting to refactor with table driven.

File: src/dragonball/src/api/v1/vmm_action.rs (hunk 4 above)
Comment (wllenyj) [reply src/dragonball/src/api/v1/vmm_action.rs]: fixed

**Comment by wllenyj** [comment]
We should support nested virtualization in Static Checking. #5378

File: .github/workflows/static-checks.yaml [hunk 6]
```diff
@@ -94,3 +94,27 @@ jobs:
...
+        run: |
+          ./ci/install_rust.sh
+          PATH=$PATH:"$HOME/.cargo/bin"
+      - name: Run Unit Test
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'force-skip-ci') }}
+        run: |
+          cd src/dragonball
+          /root/.cargo/bin/cargo version
+          rustc --version
+          # TODO: Using the cargo command directly will get `sudo: cargo: command not found` error.
```
Comment (liubin) [inline .github/workflows/static-checks.yaml:119]: Add `PATH=$PATH:"$HOME/.cargo/bin"` as line 112?

Comment (wllenyj) [reply .github/workflows/static-checks.yaml:119]: It's not work.
I have tried: `PATH=$PATH:"$HOME/.cargo/bin"` and `sudo -E PATH="$PATH" cargo test`
See 
https://github.com/kata-containers/kata-containers/actions/runs/3218377679/jobs/5262478984
https://github.com/kata-containers/kata-containers/actions/runs/3218316683/jobs/5262340478
for more details.

**Pull request merged**